xml/vk.xml

```diff
@@ -17243,16 +17293,32 @@ typedef void* <name>MTLSharedEvent_id</name>;
                 <enum bitpos="22" extends="VkImageUsageFlagBits"            name="VK_IMAGE_USAGE_RESERVED_22_BIT_EXT"/>
             </require>
         </extension>
-        <extension name="VK_INTEL_extension_272" number="272" type="device" author="INTEL" contact="Jason Ekstrand @jekstrand" supported="disabled">
-            <require>
-                <enum value="0"                                             name="VK_INTEL_EXTENSION_272_SPEC_VERSION"/>
-                <enum value="&quot;VK_INTEL_extension_272&quot;"            name="VK_INTEL_EXTENSION_272_EXTENSION_NAME"/>
-            </require>
-        </extension>
-        <extension name="VK_INTEL_extension_273" number="273" type="device" author="INTEL" contact="Jason Ekstrand @jekstrand" supported="disabled">
-            <require>
-                <enum value="0"                                             name="VK_INTEL_EXTENSION_273_SPEC_VERSION"/>
-                <enum value="&quot;VK_INTEL_extension_273&quot;"            name="VK_INTEL_EXTENSION_273_EXTENSION_NAME"/>
+        <extension name="VK_EXT_map_memory2" number="272" type="device" author="EXT" contact="Jason Ekstrand @jekstrand" supported="vulkan">
+            <require>
+                <enum value="0"                               name="VK_EXT_MAP_MEMORY_2_SPEC_VERSION"/>
+                <enum value="&quot;VK_EXT_map_memory2&quot;"  name="VK_EXT_MAP_MEMORY_2_EXTENSION_NAME"/>
+                <enum offset="0" extends="VkStructureType"    name="VK_STRUCTURE_TYPE_MEMORY_MAP_INFO_EXT"/>
+                <enum offset="1" extends="VkStructureType"    name="VK_STRUCTURE_TYPE_MEMORY_UNMAP_INFO_EXT"/>
+                <type name="VkMemoryMapInfoEXT"/>
+                <type name="VkMemoryUnmapInfoEXT"/>
+                <type name="VkMemoryUnmapFlagsEXT"/>
+                <type name="VkMemoryUnmapFlagBitsEXT"/>
+                <command name="vkMapMemory2EXT"/>
+                <command name="vkUnmapMemory2EXT"/>
```

Comment:
Do we need two functions here? If we had a single "map" function then we could make it handle map/remap/unmap rather than having map and unmap and then wondering where to put remap.

Unmapping would then just be passing in an existing mapped address with VK_NULL_HANDLE for the memory object. Remapping would be passing in an existing mapped (or reserved?) address with a different memory object/offset.

I'm not married to this idea, but I do think it would be cleaner.

Reply:
Short version: I'm trying to avoid feature creep.  The point of this extension is to make `vkMap/UnmapMemory()` extensible so that I can extend them in the next extension.  It's not to re-think all of memory mapping.

Longer: have you seen the WDDM2 intrerfaces?  They both take a handle, don't support subset maps, and don't have a remap:

https://github.com/microsoft/libdxg/blob/main/include/d3dkmthk.h#L4809

As much as I'd love it if Vulkan looked more like `mmap()` but what you're suggesting would require a drvier-side map lookup table to get from map pointers to object handles if we wanted to implement it on Windows.  Also, we'd need VUs saying you could only unmap whole memory object ranges anway so it's not like it would actually help anyone.  Remap would either be a no-op or impossible depending on how we spec'd it.

As for remap, I'm not sure exactly what semantics we'd want there.  Just grow the mapping?  On Linux, I guess we could maybe provide a "real" remap that lets you grow maps but is there actually a call for this?

Reply:
> As much as I'd love it if Vulkan looked more like mmap() but what you're suggesting would require a drvier-side map lookup table to get from map pointers to object handles if we wanted to implement it on Windows.

Ugh, Windows. Fair point. We could get around that but that's... definitely feature creepy.

> As for remap, I'm not sure exactly what semantics we'd want there. Just grow the mapping? On Linux, I guess we could maybe provide a "real" remap that lets you grow maps but is there actually a call for this?

Maybe grow, maybe set it to a completely separate memory binding, but then you run into the same issue with Windows again so blah. Windowwws...